Object Storage AWS guide

alvarosanchez opened the pull request:
Should be merged after #1078

alvarosanchez committed:
First cut of an Object Storage guide

alvarosanchez committed:
Resources for the AWS guide

alvarosanchez committed:
Simplify application

alvarosanchez committed:
Complete guide

alvarosanchez committed:
Cleanup

alvarosanchez commented:
PDF version for reviewing:

[Use the Micronaut Object Storage API to store files in AWS S3.pdf](https://github.com/micronaut-projects/micronaut-guides/files/9576429/Use.the.Micronaut.Object.Storage.API.to.store.files.in.AWS.S3.pdf)

sdelamo reviewed (changes requested):
It misses a test and a GraalVM section

I have created a PR with several improvements. I can create multiples PRs instead of a single one if you prefer that. 

https://github.com/micronaut-projects/micronaut-guides/pull/1092

sdelamo committed:
Improvements to Object Storage Guide (#1092)

Co-authored-by: Álvaro Sánchez-Mariscal <alvaro.sanchez-mariscal@oracle.com>

alvarosanchez commented:
> It misses a test and a GraalVM section

Regarding the test, if you look at 24ef20d, it had a test with Localstack, but honestly, I thought it would be more confusing, so I removed it. Moreover, for the rest of the clouds, not all the guides can be fully tested with a Testcontainers-based emulator. I tried to use `excludeTest` of `GuideMetadata.App`, but it didn't seem to work.

> I have created a PR with several improvements. I can create multiples PRs instead of a single one if you prefer that.

Thank you very much! It has been merged.

sdelamo commented:
> Regarding the test, if you look at [24ef20d](https://github.com/micronaut-projects/micronaut-guides/commit/24ef20d89ef991a08f683ce435368173642c3fad), it had a test with Localstack, but honestly, I thought it would be more confusing, so I removed it. 

We can have a test even if we don't show it in the guide text. I use guide tests as another layer of functional tests.

burtbeckwith commented on guides/micronaut-object-storage-aws/micronaut-object-storage-aws.adoc:
```diff
@@ -0,0 +1,180 @@
+external:micronaut-object-storage-base/start.adoc[]
+* An https://aws.amazon.com/[AWS account] with:
+** An IAM user with enough permissions to create and manage AWS S3 buckets.
```
remove "enough"

burtbeckwith commented on guides/micronaut-object-storage-aws/micronaut-object-storage-aws.adoc:
```diff
@@ -0,0 +1,180 @@
+
+== Create an Amazon S3 bucket
+
+First of all, we are going to
```
```suggestion
First,
```

burtbeckwith commented on guides/micronaut-object-storage-aws/micronaut-object-storage-aws.adoc:
```diff
@@ -0,0 +1,180 @@
+
+=== Delete endpoint
+
+For the delete endpoint, all we have to do is invoking the `delete` method with the expected key:
```
invoking -> invoke

burtbeckwith commented on guides/micronaut-object-storage-aws/micronaut-object-storage-aws.adoc:
```diff
@@ -0,0 +1,180 @@
+
+[NOTE]
+====
+If you want to upload a file bigger than 1MB, you need to configure:
```
bigger -> larger

burtbeckwith reviewed (changes requested):
great guide, significantly better than the one I would have written :)

alvarosanchez committed:
Add GraalVM section

alvarosanchez committed:
Address feedback

alvarosanchez committed:
Merge branch 'master' into object_storage_guides

# Conflicts:
#	buildSrc/src/main/groovy/io/micronaut/guides/GuideAsciidocGenerator.groovy

alvarosanchez committed:
Add test

sdelamo commented on guides/micronaut-object-storage-base/java/src/test/groovy/example/micronaut/AbstractProfilePicturesControllerSpec.groovy:
```diff
@@ -0,0 +1,65 @@
+        File download = client.toBlocking().retrieve(uri, File.class)
+
+        then:
+        file.text == "micronaut"
```
this is not testing the contents of the downloaded file.

sdelamo commented on guides/micronaut-object-storage-base/java/src/test/groovy/example/micronaut/AbstractProfilePicturesControllerSpec.groovy:
```diff
@@ -0,0 +1,65 @@
+        assertThatFileIsStored("alvaro.jpg", "micronaut")
+
+        when:
+        File download = client.toBlocking().retrieve(uri, File.class)
```
Instead of using `uri`, the location HTTP Header value should be used.

sdelamo replied:
I think the correct way to do this is to bind the response to a `byte[]`

sdelamo committed:
improve AWS Object Storage guide

- Change delete endpoint to use @Status(NO_CONTENT)
- Use location header in the test
- Migrate test to JUnit
- Download file as byte[]
- adds LocalStack stop as a Junit Extensions

sdelamo reviewed (changes requested):
I have created a PR which addresses the comments in this PR Review. 

https://github.com/micronaut-projects/micronaut-guides/pull/1093

Moreover, I have migrated the test to Java Junit and added the local stack feature so that the test is successfully run in CI.

alvarosanchez committed:
Fix build and cleanup

alvarosanchez committed:
Merge pull request #1093 from micronaut-projects/better-test

sdelamo committed:
Merge branch 'master' into object_storage_guides

sdelamo committed:
Object Storage

sdelamo committed:
set publication date to Sep 20th

sdelamo committed:
twitter card
